agent/php_internal_instrument.c

```diff
@@ -1574,13 +1582,20 @@ static char* nr_php_prepared_statement_make_pgsql_key(
 /*
  * Handle
  *    bool pg_close ([ resource $connection ] )
+ *    As of PHP 8.1: all resources for pg are objects instead
  */
 NR_INNER_WRAPPER(pg_close) {
   zval* pgsql = NULL;
 
+#if ZEND_MODULE_API_NO >= ZEND_8_1_X_API_NO /* PHP 8.1+ */
   if (FAILURE
       != zend_parse_parameters_ex(ZEND_PARSE_PARAMS_QUIET,
-                                  ZEND_NUM_ARGS() TSRMLS_CC, "|r", &pgsql)) {
+                                  ZEND_NUM_ARGS() TSRMLS_CC, "|o", &pgsql)) {
+#else
+  if (FAILURE
+      != zend_parse_parameters_ex(ZEND_PARSE_PARAMS_QUIET,
+                                  ZEND_NUM_ARGS() TSRMLS_CC, "|o", &pgsql)) {
```

Comment:
Should this be "|r" like the original code since this is the != PHP8.1 case?

Reply:
Fixed